hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1398,27 +1398,37 @@ private void checkOnlineRegionsReport(ServerStateNode serverNode, Set<byte[]> re
         continue;
       }
       final long lag = 1000;
-      regionNode.lock();
       try {
-        long diff = EnvironmentEdgeManager.currentTime() - regionNode.getLastUpdate();
-        if (regionNode.isInState(State.OPENING, State.OPEN)) {
-          // This is possible as a region server has just closed a region but the region server
-          // report is generated before the closing, but arrive after the closing. Make sure there
-          // is some elapsed time so less false alarms.
-          if (!regionNode.getRegionLocation().equals(serverName) && diff > lag) {
-            LOG.warn("Reporting {} server does not match {} (time since last "
-              + "update={}ms); closing...", serverName, regionNode, diff);
-            closeRegionSilently(serverNode.getServerName(), regionName);
-          }
-        } else if (!regionNode.isInState(State.CLOSING, State.SPLITTING)) {
-          // So, we can get report that a region is CLOSED or SPLIT because a heartbeat
-          // came in at about same time as a region transition. Make sure there is some
-          // elapsed time so less false alarms.
-          if (diff > lag) {
-            LOG.warn("Reporting {} state does not match {} (time since last update={}ms)",
-              serverName, regionNode, diff);
+        // It is likely that another thread is currently holding the lock. To avoid deadlock, use
+        // tryLock waiting for a specified period of time
+        if (regionNode.tryLock(10, TimeUnit.SECONDS)) {
```

Comment:
tryLock should be placed outside the try block, as we do not need to unlock if we get a false here.

Reply:
@Apache9 Thank you for your suggestion. The position of the tryLock has been adjusted.

Reply:
Maybe we could just use tryLock without any timeout? Then it will not throw any exceptions so we do not need this two level try catch block. For me I think it is OK as this is not a critical check.

Reply:
This is a very good suggestion, and it has been adjusted accordingly.